src/tokenize.c

```diff
@@ -72,6 +69,73 @@ static char *DefaultNormalize(char *s, char *dst, size_t *len) {
   return dst;
 }
 
+static char *DefaultNormalize_utf8(char *s, char *dst, size_t *len) {
+  // TODO: set locale depending on the index language?
+  setlocale(LC_ALL, "en_US.UTF-8");
+
+  size_t origLen = *len;
+  char *realDest = s;
+  size_t dstLen = 0;
+
+#define SWITCH_DEST()        \
+  if (realDest != dst) {     \
+    realDest = dst;          \
+    memcpy(realDest, s, ii); \
+  }
+  // set to 1 if the previous character was a backslash escape
+  int escaped = 0;
+  mbstate_t state;
+  memset(&state, 0, sizeof(state));
+  wchar_t wc;
+  size_t len_wc;
+
+  for (size_t ii = 0; ii < origLen;) {
+    len_wc = mbrtowc(&wc, &s[ii], MB_CUR_MAX, &state);
+    if (len_wc == (size_t)-1 || len_wc == (size_t)-2) {
+      // Handle invalid multi-byte sequence
+      ii++;
+      continue;
+    }
+
+    if (iswupper(wc)) {
+      SWITCH_DEST();
+      wchar_t lower_wc = towlower(wc);
+      wcrtomb(&realDest[dstLen], lower_wc, &state);
+      dstLen += len_wc;
+    } else if ((iswblank(wc) && !escaped) || iswcntrl(wc)) {
+      SWITCH_DEST();
+    } else if (wc == L'\\' && !escaped) {
+      SWITCH_DEST();
+      escaped = 1;
+      ii += len_wc;
+      continue;
+    } else {
+      memcpy(&dst[dstLen], &s[ii], len_wc);
+      dstLen += len_wc;
+    }
+    ii += len_wc;
+    escaped = 0;
+  }
+
+  *len = dstLen;
+  return dst;
+}
+
+/**
+ * Normalizes text.
+ * - s contains the raw token
+ * - dst is the destination buffer which contains the normalized text
+ * - len on input contains the length of the raw token. on output contains the
+ * on output contains the length of the normalized token
+ */
+static char *DefaultNormalize(char *s, char *dst, size_t *len) {
+  if (setlocale(LC_ALL, "en_US.UTF-8") != NULL) {
```

Comment:
Why would this not succeed (according to the docs)?
I think we should raise an error in that case, in order to have consistent behavior and not switch behaviors under the hood without the user knowing why (and us probably, as there is no log as well)

Reply:
It will fail if the selected locale is not found. The OS could not include it.
I'm not sure if we should raise an error or maybe add a configuration param to let the user to decide to use multi-byte or not.

Reply:
I think consistency is key here. We can't produce different results depending on the OS